src/agent/adu_core_interface/src/adu_core_interface.c

```diff
@@ -820,6 +820,7 @@ _Bool AzureDeviceUpdateCoreInterface_ReportStateAndResultAsync(
     }
 
     jsonString = json_serialize_to_string(rootValue);
+    json_value_free(rootValue);
```

Comment:
Thanks for catching this! 
Do you mind move this to line 840 please? In case we need to add some code that uses need rootValue before **done:**